src/RulesEngine/Models/Workflow.cs

```diff
@@ -1,25 +1,41 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using System;
 using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
 
 namespace RulesEngine.Models
 {
+    [Obsolete("WorkflowRules class is deprecated. Use Workflow class instead.")]
+    [ExcludeFromCodeCoverage]
+    public class WorkflowRules : Workflow {
+    }
+
     /// <summary>
     /// Workflow rules class for deserialization  the json config file
     /// </summary>
     [ExcludeFromCodeCoverage]
-    public class WorkflowRules
+    public class Workflow
     {
+        /// <summary>
+        /// Gets the workflow Id.
+        /// </summary>
+        public int Id { get; set; }
+
```

Comment:
@alexreich, I would like to know your thought process behind adding `Id` in workflow class. If it is only for EF related usage, we can add it to a class extending Workflow and place it in efcore demo. 

We generally try to avoid putting additional fields in workflow or rule as it might get confusing for a user to add more fields.

If we really need to add additional fields at workflow level, I recommend adding `Properties` field , similar to what we have in Rule :- 
https://github.com/microsoft/RulesEngine/blob/f9def1c6f738662828d6271b8e3cef431495ed1d/src/RulesEngine/Models/Rule.cs#L22-L28

Then you could add `id` or any other fields as part of the properties
e.g. `workflow.Properties = new Dictionary<string,object> {  {"id", 124} }`

Let me know your thoughts on this

Reply:
@abbasc52 Extending Workflow won't work per the error below - an entirely new class would have to be created and mapped. I've added (OPTIONAL) to the summary decoration to prevent confusion for users adding fields. Let me know your thoughts.

`System.InvalidOperationException: 'A key cannot be configured on 'DatabaseRule' because it is a derived type. The key must be configured on the root type 'Rule'. If you did not intend for 'Rule' to be included in the model, ensure that it is not referenced by a DbSet property on your context, referenced in a configuration call to ModelBuilder, or referenced from a navigation on a type that is included in the model.'`

Adding the Key to property also can't be done since it's not a "simple property or field access":

`entity.HasKey(k => k.Properties["id"]);
entity.Property(p => p.Properties["id"]).ValueGeneratedOnAdd();`

results in the following:

`System.ArgumentException: 'The expression 'k => k.Properties.get_Item("id")' is not a valid member access expression. The expression should represent a simple property or field access: 't => t.MyProperty'. When specifying multiple properties or fields, use an anonymous type: 't => new { t.MyProperty, t.MyField }'. (Parameter 'memberAccessExpression')'`

Reply:
@abbasc52 We can but the issue is that you can't change the key (in this case the WorkflowName) because it's the Primary Key. This issue is what led me to the whole "Id odyssey" to begin with. I've been working on a RulesEngineEditor in Blazor and ran into this issue - the work around would be to delete the whole Workflow tree and re-insert it for a simple rename.

Reply:
@alexreich, I still have concerns related to id.

Firstly, it is not used anywhere (when not using EF) in the RulesEngine so expecting users to provide it would be a bad practice.

Secondly, for users who don't provide it, it will be all set to default value, which in this case beats the purpose of an id.

Thirdly, RulesEngine provides no validation for id, nothing prevents user from duplicating ids as of now.

We need to tackle these problems before we can introduce this field as people will have questions regarding the same.

RulesEngine models are not meant to be EF models directly and there can be variations. I would prefer keeping them as loosely coupled as possible.
RulesEngine should be pluggable to any data source and its design should not be dictated by some source.

I believe having a mapping between EFcore Entity classes and RulesEngine class makes more sense as not all users will use RulesEngine with EF. 

If needed, we can expose all the additional setup and classes as nuget addons. But i would like to avoid it being part of the core library.

Reply:
Reverted Id.

Using a mapping class just for EF (and keeping up w/schema changes) would bring more to maintain and moving away from DRY, even w/using a helper like Automapper.

Will work around issues w/renaming Workflow and Rule.